pkg/kube/util/reference/secret_references.go

```diff
@@ -1,27 +1,13 @@
 package reference
 
 import (
-	"context"
-
-	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
-	crc "sigs.k8s.io/controller-runtime/pkg/client"
 
-	ejv1 "code.cloudfoundry.org/quarks-job/pkg/kube/apis/extendedjob/v1alpha1"
+	qjv1a1 "code.cloudfoundry.org/quarks-job/pkg/kube/apis/quarksjob/v1alpha1"
 )
 
-// GetSecretsReferencedBy returns a list of all names for Secrets referenced by the object
-// The object can be an ExtendedStatefulSet, an ExtendedJob or a BOSHDeployment
-func GetSecretsReferencedBy(ctx context.Context, client crc.Client, object interface{}) (map[string]bool, error) {
-	switch object := object.(type) {
-	case ejv1.ExtendedJob:
-		return getSecretRefFromEJob(object), nil
-	default:
```

Comment:
Personally I like the `default`, to know what could have gone wrong(e.g. in this case). But I also think is not needed here, while this `func` is explicitly only use by the quarksjob.